types/aws-lambda/trigger/kinesis-stream.d.ts

```diff
@@ -28,7 +31,20 @@ export interface KinesisStreamEvent {
     Records: KinesisStreamRecord[];
 }
 
-// https://docs.aws.amazon.com/lambda/latest/dg/with-kinesis.html
+// https://docs.aws.amazon.com/lambda/latest/dg/with-kinesis.html#services-kinesis-windows
+export interface KinesisStreamTumblingWindowEvent {
+    window: { start: string; end: string };
+    state?: { [key: string]: any };
+    isFinalInvokeForWindow: boolean;
+    isWindowTerminatedEarly: boolean;
+    Records: KinesisStreamRecord[];
+}
+
+export interface KinesisStreamStateResponse {
```

Comment:
Extend `KinesisStreamBatchResponse` as `batchItemFailures` is missing from the definition.

Reply:
According to [docs](https://docs.aws.amazon.com/lambda/latest/dg/with-kinesis.html#services-kinesis-windows), this should be the response object:
```js
{
    "state": {
        "1": 282,
        "2": 715
    },
    "batchItemFailures": []
}
```
Please, add `batchItemFailures` by extending `KinesisStreamBatchResponse`. Or, if it's optional, support that with a citation from docs.

Reply:
I extended via `extends Partial<KinesisStreamBatchResponse>` since when returning a state, returning `batchItemFailures: []` is optional. Do you agree?